ganga/GangaCore/Utility/Monitoring.py

```diff
@@ -0,0 +1,54 @@
+from GangaCore.Utility.Config import getConfig
+
+import os
+from memory_profiler import profile
+import cProfile
+import time
+
+timestr = time.strftime("-%Y%m%d-%H%M%S")
+
+c = getConfig('Configuration')
+
+path = os.path.join(c['gangadir'],'logs')
+
+def makedir(path):
+    try:
+        if not os.path.exists(path):
+            os.makedirs(path)
+        return path
+    except Exception as e:
+        pass
+    # Will do error handling later
+
+makedir(path)
+
+def cpu_profile(func):
+    mpath = makedir(os.path.join(path,'cpu_profile/'))
+    def wrapper(*args, **kwargs):
+        datafn = mpath+func.__name__ +timestr+".profile"
+        prof = cProfile.Profile()
+        retval = prof.runcall(func, *args, **kwargs)
+        prof.dump_stats(datafn)
+        return retval
+    return wrapper
+
+def cpu_profiler(cls, profile_cpu=c['Profile_CPU']):
+    if profile_cpu:
+        for key, value in list(vars(cls).items()):
+            if callable(value):
+                setattr(cls, key, cpu_profile(value))
+    else:
+        pass
+    return cls
+
+def mem_profiler(cls, profile_memory=c['Profile_Memory']):
+    if profile_memory:
+        cpath = makedir(os.path.join(path,'memory_profiles/'))
+        file_name = cpath+cls.__name__+timestr+'.log'
+        fp = open(file_name, 'w+')
+        for key, value in list(vars(cls).items()):
+            if callable(value):
+                setattr(cls, key, profile(value, stream=fp, precision=6))
+    else:
+        pass
+    return cls
```

Comment:
Add the missing newline.

Reply:
I have updated the code to meet PEP 8 requirements.